### ISSUE TITLE
No ex assign seems buggy

### ISSUE THREAD

**Issue by blakeembrey** [buzinas/tslint-eslint-rules#85]
Enabled it and broke builds across at least a dozen repos. Trying to do clean up now on it, but here's some examples (none do any exception assignment):

* https://travis-ci.org/TypeStrong/ts-node/builds/131967778
* https://travis-ci.org/typings/typings/builds/131969123
* https://travis-ci.org/blakeembrey/promise-finally/builds/131968888 (this one is a _very_ trivial repro)

Trivial replication from `tslint-config-standard`:

```ts
const y = 10

if (y === 0) {
  function test () {}
}
```

**Comment by buzinas** [comment]
There is already an issue opened for `no-ex-assign`, there are lots of false positives.

Unfortunately, I don't have time to look at this now, but if you want, please, feel free to open a pull request for that.

I'll close this one to keep the other opened: https://github.com/buzinas/tslint-eslint-rules/issues/54
